tests/probes_test.go

```diff
@@ -26,130 +26,117 @@ var _ = Describe("[ref_id:1182]Probes", func() {
-				Expect(err).ToNot(HaveOccurred())
-				return vmiReady(vmi)
-			}, 60, 1).Should(Equal(v1.ConditionTrue))
+			Eventually(isVMIReady, 60, 1).Should(Equal(true))
 			Expect(tests.PodReady(tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault))).To(Equal(v1.ConditionTrue))
 		},
-			table.Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server", tcpProbe, startTCPServer),
+			table.Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server", tcpProbe, tests.StartTCPServer),
 			table.Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server", httpProbe, tests.StartHTTPServer),
 		)
 
 		table.DescribeTable("should fail", func(readinessProbe *v12.Probe) {
 			By("Specifying a VMI with a readiness probe")
-			vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
-			vmi.Spec.ReadinessProbe = readinessProbe
-			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
-			Expect(err).ToNot(HaveOccurred())
-			// It may come to modify retries on the VMI because of the kubelet updating the pod, which can trigger controllers more often
-			tests.WaitForSuccessfulVMIStartIgnoreWarnings(vmi)
+			vmi = createReadyCirrosVMIWithReadinessProbe(virtClient, readinessProbe)
 
-			Expect(tests.PodReady(tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault))).To(Equal(v1.ConditionFalse))
-			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &v13.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			Expect(vmiReady(vmi)).To(Equal(v1.ConditionFalse))
+			// pod is not ready until our probe contacts the server
+			assertPodNotReady(virtClient, vmi)
 
 			By("Checking that the VMI and the pod will consistently stay in a not-ready state")
-			Consistently(func() v1.ConditionStatus {
-				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &v13.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				return vmiReady(vmi)
-			}, 60, 1).Should(Equal(v1.ConditionFalse))
+			Consistently(isVMIReady).Should(Equal(false))
 			Expect(tests.PodReady(tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault))).To(Equal(v1.ConditionFalse))
 		},
 			table.Entry("[test_id:1220][posneg:negative]with working TCP probe and no running server", tcpProbe),
 			table.Entry("[test_id:1219][posneg:negative]with working HTTP probe and no running server", httpProbe),
 		)
-	})
 
-	Context("for liveness", func() {
-
-		tcpProbe := &v12.Probe{
-			PeriodSeconds:       5,
-			InitialDelaySeconds: 90,
-			Handler: v12.Handler{
+		table.DescribeTable("should succeed probing an IPv6 IP on dual stack clusters", func(readinessProbe *v12.Probe) {
+			if !flags.IsDualStackedNetworkConfig {
+				Skip("This test requires a dual stack network config.")
+			}
```

Comment:
The --dual-stack flag is not for skip tests, is to check that the cluster is dual stack at a pre-flight (maybe at BeforeSuite or outsider script). This test hast to skip if there is no IPv6 at IPs.

Reply:
I've replied to this in https://github.com/kubevirt/kubevirt/pull/4129#discussion_r486135073. 

Let's open this can of worms :)

Reply:
OK; replying here for transparency.

We have instead moved to a solution where we check a well known pod (I'm looking at you, virt-handler) if it has an IPv6 address.